### DISCUSSION TITLE
problem with charaters  >127

### DISCUSSION THREAD

**Discussion by Karl-250** [olikraus/u8g2#2380] [Q&A]
Hello everyone,
While getting familiar with the u8g2 library, I am using a small monochrome OLED Display 0,91"with I2C Interface and the ESP32 chip with the ESP-IDF. The project I programmed was a progress bar with charaters that are inverted when the progress bar overlay them. I used the DrawColor Option 2 -> XOR and the font in transparent mode. Everything worked graet !! until the moment I added "°C" to the string. ° and C are flickering. Using a character <=127 in the font list (u8g2_font_helvB24_tf) there is no problem. Using characters >127 the flickering occurs. For displaing characters > 127 I use „u8g2_DrawUTF8“.
Interesting is, when I change the sequence from „°C“ to „C°“, the „C“ is displayed correctly and "°" flickers. Is that a known bug ? Does there exists a workaround ?

Best regards Karl

**Comment by olikraus** [comment]
Difficult to tell without the code, but maybe https://github.com/olikraus/u8g2/wiki/u8g2reference#enableutf8print helps

**Comment by Karl-250** [comment]
Hi Oli,

Here the relevant code snippet:

void ProgressBar(int wert, int last_wert)
{
	u8g2_SetFontMode(&u8g2,1);//transparent mode
	
	if(wert<10) //right adjust digits
		{
			OffsetDigit = 18*2;
		}
		else if(wert>=10 && wert<100)
		{
			OffsetDigit = 18;
		}
		else
		{
			OffsetDigit = 0;
		}

		//assemble string to be displayed
		sprintf(value_as_c, "%d",wert);//convert numerical value to string
		strcat(value_as_c, "°C");// add to string at the end

		printf("%s\n",value_as_c);//print to monitor

		//clear old characters and box
		u8g2_SetDrawColor(&u8g2,0);
		u8g2_DrawStr(&u8g2,(Position+Last_OffsetDigit),28,last_value_as_c);
		u8g2_DrawBox(&u8g2,3,3,last_wert,26);

		//draw actual BOX
		u8g2_SetDrawColor(&u8g2,1);	 		
		u8g2_DrawBox(&u8g2,3,3,wert,26);

		//draw actual string
		u8g2_SetDrawColor(&u8g2,2);//Mode XOR
		u8g2_DrawUTF8(&u8g2,(Position+OffsetDigit),28,value_as_c);

		u8g2_SendBuffer(&u8g2);

		Last_OffsetDigit = OffsetDigit;								
		strcpy(last_value_as_c,value_as_c);		
		
}

When I change the sequence of " °C° to " C°", the C is displayed correctly.
The command "enableutf8print" only exists with Arduino, I am using the ESP-IDF.

best regards Karl

**Comment by Karl-250** [comment]
Hi,
I found the solution:
For drawing I used "DrawUTF8" for clearing the "DrawString" command. For clearing its also neccessary to use the DrawUTF8 command .
best regards Karl

**Reply by olikraus** [reply]
Yes... correct.
